### PR TITLE
Organized and updated spatial shader docs

### LINKED ISSUE #2606: Add "Shadow to Opacity" to Spatial Shader reference
https://github.com/godotengine/godot/pull/29031 Adds a new render mode, ``shadow_to_opacity``, and a new built-in, ``ALPHA``, to the light processor function. Both should be added to the [Spatial Shader doc](http://docs.godotengine.org/en/latest/tutorials/shading/shading_reference/spatial_shader.html)

_edit:_ Oh, and https://github.com/godotengine/godot/pull/24487 added ``UV`` and ``UV2`` to the the light processor function as well. So those two built-ins should be added.

### PR DESCRIPTION
**Pull request by vortexofdoom**
Fixes #2606. Also accomplishes a sub-portion of #2605, but I wanted to submit this first to make sure the organization is actually good. It's definitely more organized and consistent throughout, but there's probably still room for improvement. The grouping is probably more improved than the overall order.

### CODE REVIEW

File: tutorials/shading/shading_reference/spatial_shader.rst [hunk 1]
```diff
@@ -72,6 +72,8 @@ Render modes
 +---------------------------------+-----------------------------------------------------------------------+
 | **ambient_light_disabled**      | Disable contribution from ambient light and radiance map.             |
 +---------------------------------+-----------------------------------------------------------------------+
+| **shadow_to_opacity**           | Allows for overlaying shadows onto a camera feed (good for AR use).   |
```
Comment (clayjohn) [inline tutorials/shading/shading_reference/spatial_shader.rst]: Im not sure this description properly captures what shadow_to_opacity does. It needs to be more descriptive about how this renders non-shadowed areas transparent with shadowed areas opaque.

**Comment by vortexofdoom** [comment]
"Lighting modifies the alpha so shadowed areas are opaque and non-shadowed areas are transparent. Useful for overlaying shadows onto a camera feed in AR."

Would that be better? A little bit on the nose with regard to your requested changes, but you probably said it as well as it could be said. I think I was trying to avoid getting too wordy, but most of the descriptions are so ridiculously terse I probably shouldn't have used them as a guideline. I was only going to squash changes once so I thought I'd ask first to make sure I didn't need to change it again.

**Comment by clayjohn** [comment]
> Lighting modifies the alpha so shadowed areas are opaque and non-shadowed areas are transparent. Useful for overlaying shadows onto a camera feed in AR

Thats great! Descriptive explanations are definitely preferred. And its great to let people know the intended use case when there really is only one. Don't worry about the length.

**Comment by akien-mga** [comment]
Thanks!
